exchanges/graphcache/src/store/store.ts

```diff
@@ -170,18 +170,27 @@ export class Store<
 
   invalidate(entity: Entity, field?: string, args?: FieldArgs) {
     const entityKey = this.keyOfEntity(entity);
+    const shouldInvalidateType =
```

Comment:
I wonder if we should narrow this down to only supporting `invalidate({ __typename: 'Todo' })` as this explicitly can't derive a key nor be a global-id which can be the case with `invalidate('x')`